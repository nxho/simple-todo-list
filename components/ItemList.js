import React from 'react';
import {
	ScrollView,
	StyleSheet,
	Text,
	TextInput,
	View,
} from 'react-native';
import Checkbox from './Checkbox';

export default class ItemList extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return(
			<ScrollView
				style={styles.container}
				contentContainerStyle={styles.contentContainer}
				pinchGestureEnabled={false}
				bounces={false}
				showsVerticalScrollIndicator={false}
			>
				{
					this.props.items.map((item, index) => {
						return(
							<View key={index} style={styles.itemContainer}>
								<TextInput
									testID={`item${index}-text-input`}
									style={[styles.textInput, item.checked ? styles.checkedText : null]}
									onChangeText={(text) => this.props.onChange(text, index)}
									value={item.text}
								/>
								<Checkbox
									style={styles.checkbox}
									onCheckItem={() => this.props.onCheckItem(index)}
									checked={item.checked}
								/>
							</View>
						);
					})
				}
			</ScrollView>
		);
	}
}

const styles = StyleSheet.create({
	container: {
	},
	contentContainer: {
		justifyContent: 'flex-start',
	},
	itemContainer: {
		height: 50,
		flexDirection: 'row',
		justifyContent: 'flex-start',
	},
	textInput: {
		flex: 6,
		color: 'black',
	},
	checkedText: {
		color: '#EDEFF4',
	},
	checkbox: {
		flex: 1,
	},
});

