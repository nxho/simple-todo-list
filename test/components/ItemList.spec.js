import React from 'react';
import { shallow } from 'react-native-testing-library';
import ItemList from '../../components/ItemList';

describe('ItemList component', () => {
  it('renders correctly', () => {
    const items = [];
    const mockFn = jest.fn();
    const { output } = shallow(<ItemList items={items} onChange={mockFn} />);

    expect(output).toMatchSnapshot();
  });
});

