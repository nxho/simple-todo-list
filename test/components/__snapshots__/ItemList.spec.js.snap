// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component renders correctly 1`] = `
<ScrollViewMock
  bounces={false}
  contentContainerStyle={
    Object {
      "justifyContent": "flex-start",
    }
  }
  pinchGestureEnabled={false}
  showsVerticalScrollIndicator={false}
  style={Object {}}
/>
`;
